e2e/specs/st_set_page_config.spec.js

```diff
@@ -51,4 +51,47 @@ describe("st.set_page_config", () => {
       "wide-mode"
     );
   });
+
+  describe("double-setting set_page_config", () => {
+    beforeEach(() => {
+      // Rerun the script to ensure a fresh slate.
+      // This is done by typing r on the page
+      cy.get("body").type("r");
+      // Ensure the rerun completes
+      cy.wait(1000);
+    });
+
+    it("should not display an error when st command is used in a callback", () => {
+      cy.get(".stButton button")
+        .eq(1)
+        .click();
+
+      cy.get(".stException").should("not.exist");
+      cy.title().should("eq", "Heya, world?");
+    });
+
+    it("Should display an error when st.set_page_config is called multiple times in a callback", () => {
+      cy.get(".stButton button")
+        .eq(2)
+        .click();
+
+      cy.get(".stException")
+        .contains("set_page_config() can only be called once per app")
+        .should("exist");
+      // Ensure that the first set_page_config worked
+      cy.title().should("eq", "Change 1");
+    });
+
+    it("Should display an error when st.set_page_config is called multiple times in a callback", () => {
```

Comment:
nit: should this test title be changed to "is called after being called in a callback?"